spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/ServiceInstanceChooser.java

```diff
@@ -33,4 +34,13 @@
 	 */
 	ServiceInstance choose(String serviceId);
 
+	/**
+	 * Chooses a ServiceInstance from the LoadBalancer for the specified service and
+	 * LoadBalancer request.
+	 * @param serviceId The service ID to look up the LoadBalancer.
+	 * @param request The request to pass on to the LoadBalancer
+	 * @return A ServiceInstance that matches the serviceId.
+	 */
+	<T> ServiceInstance choose(String serviceId, Request<T> request);
```

Comment:
I'm not sure this method is required. Should call execute?

Reply:
From what I see, this entire interface seems to have been added specifically for the RetryInterceptor; I can have a closer look at that interceptor and see if it can be reworked in such a way that this interface can be removed, but I'd rather do it as a separate issue and PR.